### PR TITLE
Allow comparing frame with data to SkyCoord

### LINKED ISSUE #13476: Comparing Frame with data and SkyCoord with same data raises exception
### Description

`SkyCoord` instances and `Frame` instances with data are somewhat used interchangebly and I am still not sure after all this time spending with astropy what is preferable when...

So it's  a bit surprising to me, that comparing a frame with data to a `SkyCoord` instance with exactly the same data raises an exception:

```

### Expected behavior
<!-- What did you expect to happen. -->
Compare to true / false depending on data.

### Actual behavior
Exception


### Steps to Reproduce


```python
In [1]: from astropy.coordinates import SkyCoord, ICRS

In [2]: frame = ICRS("0d", "0d")

In [3]: coord = SkyCoord(frame)

In [4]: frame == coord
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [4], in <cell line: 1>()
----> 1 frame == coord

File ~/.local/lib/python3.10/site-packages/astropy/coordinates/baseframe.py:1657, in BaseCoordinateFrame.__eq__(self, value)
   1651 def __eq__(self, value):
   1652     """Equality operator for frame.
   1653 
   1654     This implements strict equality and requires that the frames are
   1655     equivalent and that the representation data are exactly equal.
   1656     """
-> 1657     is_equiv = self.is_equivalent_frame(value)
   1659     if self._data is None and value._data is None:
   1660         # For Frame with no data, == compare is same as is_equivalent_frame()
   1661         return is_equiv

File ~/.local/lib/python3.10/site-packages/astropy/coordinates/baseframe.py:1360, in BaseCoordinateFrame.is_equivalent_frame(self, other)
   1358     return True
   1359 elif not isinstance(other, BaseCoordinateFrame):
-> 1360     raise TypeError("Tried to do is_equivalent_frame on something that "
   1361                     "isn't a frame")
   1362 else:
   1363     return False

TypeError: Tried to do is_equivalent_frame on something that isn't a frame

```

### PR DESCRIPTION
**Pull request by maxnoe**
### Description

Fixes #13476 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/coordinates/tests/test_frames.py
Comment (pllim) [inline astropy/coordinates/tests/test_frames.py:1547]: This is unnecessary.
```suggestion
        frame == SkyCoord(AltAz("0d", "1d"))
```

**Review by pllim — COMMENTED** [review]
Need a change log. Thanks!

Since this is also behavior change, maybe shouldn't backport?

**Comment by pllim** [comment]
p.s. So many coordinates maintainers listed. Pinging all of them seem overkill, so I randomly picked two. Feel free to ping specific people to review as needed. Thanks!

**Comment by mhvk** [comment]
@maxnoe - in principle OK, but I think the logic should be reversed: the frames do not really know about `SkyCoord`, while `SkyCoord` knows about frames, so my sense would be that `BaseCoordinateFrame.__eq__` should return `NotImplemented` on anything that is not a `BaseCoordinateFrame`, while `SkyCoord.__eq__` should turn a `BaseCoordinateFrame` instance into a `SkyCoord` and then compare. This has the advantage that if the `SkyCoord` has extra attributes, which are not on the frame, then equality will still (correctly) return `False`.

**Comment by maxnoe** [comment]
That also solves the circular import problem, nice

**Review by eerovaher — CHANGES_REQUESTED** [review]
This is a good idea, but I'd like there to be a test for something like a `FK5` frame and a `SkyCoord` also in `FK5`, but with a different equinox.

File: astropy/coordinates/tests/test_frames.py
Comment (maxnoe) [reply astropy/coordinates/tests/test_frames.py:1522]: It silences my linter from a "Unused expression warning"...

**Comment by maxnoe** [comment]
> This is a good idea, but I'd like there to be a test for something like a FK5 frame and a SkyCoord also in FK5, but with a different equinox.

This currently raises an error (must be in an equivalent frame). I guess this is the desired behavior, or should it just return `False`?

**Comment by maxnoe** [comment]
The PR is labeled "bug", however, it seems this is more of a new feature (I put the changelog entry as feature).

@eerovaher Should I change the changelog entry or do we change the label?

File: astropy/coordinates/tests/test_frames.py
Comment (eerovaher) [inline astropy/coordinates/tests/test_frames.py:1516]:
```suggestion
    other = SkyCoord(ICRS(0 * u.deg, 1 * u.deg))
```

**Comment by pllim** [comment]
So is the original issue this is fixing not actually a bug? I removed the label here.

**Review by eerovaher — CHANGES_REQUESTED** [review]
I think this can be merged after one more update, and I'd also like there to be only a single commit.

File: astropy/coordinates/sky_coordinate.py
Comment (eerovaher) [inline astropy/coordinates/sky_coordinate.py:383]: This case should be covered with a test.

Comment (maxnoe) [reply astropy/coordinates/sky_coordinate.py:383]: done

**Review by eerovaher — APPROVED** [review]
Using strings as angle values instead of `Quantity` instances adds around a couple hundred microseconds to the test execution time, but I'm willing to merge this as is.

**Comment by mhvk** [comment]
OK, let's merge! Thanks, @maxnoe -- also for another nice exchange where we ended up on a simple solution!

**Pull request merged**
